data_source_transip_private_network.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/transip/gotransip/v6/repository"
+	"github.com/transip/gotransip/v6/vps"
+)
+
+func dataSourcePrivateNetwork() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcePrivateNetworkRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Required: true,
+				Type:     schema.TypeString,
+				ForceNew: true,
+			},
+			"description": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"is_blocked": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"is_locked": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"vps_names": {
+				Computed: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePrivateNetworkRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(repository.Client)
+	repository := vps.PrivateNetworkRepository{Client: client}
+	name := d.Get("name").(string)
+	// getPrivateNetworkID(d, m)
+
+	p, err := repository.GetByName(name)
+	if err != nil {
+		return fmt.Errorf("failed to lookup private network %q: %s", name, err)
+	}
+
+	log.Printf("[DEBUG] terraform-provider-transip success: private network %s found, getting details.\n", name)
```

Comment:
I would reserve logging for stuff that still requires debugging sometimes, like complex logical statements or unreliable API responses. This seems straightforward code so all information you get from the debug log you already get when running the code as well, making the debug statement superfluous.